Frameworks/CoreText/CTLine.mm

```diff
@@ -22,25 +22,89 @@
 #import <CoreText/CTTypesetter.h>
 #import <CoreFoundation/CFString.h>
 
-#include <memory>
+#import <memory>
 #import <algorithm>
 #import <numeric>
 #import <cwchar>
 #import <vector>
 
+static inline CFRange __CFRangeIntersectionRange(CFRange a, CFRange b) {
+    CFIndex start = std::max(a.location, b.location);
+    CFIndex end = std::min(a.location + a.length, b.location + b.length);
+    if (end <= start) {
+        return CFRangeMake(0, 0);
+    }
+    return CFRangeMake(start, end - start);
+}
+
+static void __CFAtributedStringAppendAttributedString(CFMutableAttributedStringRef attString, CFAttributedStringRef appendString) {
+    CFRange replaceRange = { CFAttributedStringGetLength(attString), 0 };
+    CFAttributedStringReplaceString(attString, replaceRange, CFAttributedStringGetString(appendString));
+
+    CFRange enumerationRange = { 0, CFAttributedStringGetLength(appendString) };
+    CFIndex index = enumerationRange.location;
+    CFIndex prevlen = CFAttributedStringGetLength(attString);
+    CFIndex lenDiff = 0;
+    CFRange currentRange;
+
+    while ((index >= enumerationRange.location) && index < (enumerationRange.location + enumerationRange.length)) {
```

Comment:
this reflects what NSAttributedString is doing.
Given i don't have the full range of context on the CoreText dancing here, I decide to do the same.
which is also pretty trivial. If you are 100% this is sufficient, i can move forward. 

---
In reply to: [112734332](https://github.com/Microsoft/WinObjC/pull/2586#discussion_r112734332) [](ancestors = 112734332)